src/clp_ffi_js/ir/StructuredIrUnitHandler.cpp

```diff
@@ -0,0 +1,156 @@
+    return clp::ffi::ir_stream::IRErrorCode::IRErrorCode_Success;
+}
+
+auto StructuredIrUnitHandler::handle_schema_tree_node_insertion(
+        clp::ffi::SchemaTree::NodeLocator schema_tree_node_locator
+) -> clp::ffi::ir_stream::IRErrorCode {
+    ++m_current_node_id;
+
+    auto const& key_name{schema_tree_node_locator.get_key_name()};
+    if (m_log_level_key == key_name) {
+        m_log_level_node_id.emplace(m_current_node_id);
+    } else if (m_timestamp_key == key_name) {
+        m_timestamp_node_id.emplace(m_current_node_id);
+    }
+
+    return clp::ffi::ir_stream::IRErrorCode::IRErrorCode_Success;
+}
+
+auto StructuredIrUnitHandler::handle_end_of_stream() -> clp::ffi::ir_stream::IRErrorCode {
+    return clp::ffi::ir_stream::IRErrorCode::IRErrorCode_Success;
+}
+
+auto StructuredIrUnitHandler::get_log_level(
+        StructuredLogEvent::NodeIdValuePairs const& id_value_pairs
+) const -> LogLevel {
+    LogLevel log_level{LogLevel::NONE};
+
+    if (false == m_log_level_node_id.has_value()) {
+        return log_level;
+    }
+
+    auto const& log_level_pair{id_value_pairs.at(m_log_level_node_id.value())};
+
+    if (false == log_level_pair.has_value()) {
+        return log_level;
+    }
+
+    if (log_level_pair->is<std::string>()) {
+        auto const& log_level_node_value = log_level_pair.value().get_immutable_view<std::string>();
+        log_level = parse_log_level(log_level_node_value);
+    } else if (log_level_pair->is<clp::ffi::value_int_t>()) {
+        auto const& log_level_node_value
+                = (log_level_pair.value().get_immutable_view<clp::ffi::value_int_t>());
+        if (log_level_node_value >= clp::enum_to_underlying_type(cValidLogLevelsBeginIdx)
+            && log_level_node_value < clp::enum_to_underlying_type(LogLevel::LENGTH))
+        {
+            log_level = static_cast<LogLevel>(log_level_node_value);
+        }
+    } else {
+        auto log_event_idx = m_deserialized_log_events->size();
+        SPDLOG_ERROR("Log level type is not int or string for log event index {}", log_event_idx);
+    }
+
+    return log_level;
+}
+
+auto StructuredIrUnitHandler::get_timestamp(
+        StructuredLogEvent::NodeIdValuePairs const& id_value_pairs
+) const -> clp::ffi::value_int_t {
+    clp::ffi::value_int_t timestamp{0};
+
+    if (false == m_timestamp_node_id.has_value()) {
+        return timestamp;
+    }
+
+    auto const& timestamp_pair{id_value_pairs.at(m_timestamp_node_id.value())};
+
+    if (false == timestamp_pair.has_value()) {
+        return timestamp;
+    }
+
+    if (timestamp_pair->is<clp::ffi::value_int_t>()) {
+        timestamp = timestamp_pair.value().get_immutable_view<clp::ffi::value_int_t>();
+    } else {
+        // TODO: Add support for parsing timestamp values of string type.
+        auto log_event_idx = m_deserialized_log_events->size();
+        SPDLOG_ERROR("Timestamp type is not int for log event index {}", log_event_idx);
+    }
```

Comment:
more of a preference. feel free to disagree

```suggestion
    auto const& timestamp_pair{id_value_pairs.at(m_timestamp_node_id.value())};
    if (false == timestamp_pair.has_value()) {
        return timestamp;
    }
    if (timestamp_pair->is<clp::ffi::value_int_t>()) {
        timestamp = timestamp_pair.value().get_immutable_view<clp::ffi::value_int_t>();
    } else {
        // TODO: Add support for parsing timestamp values of string type.
        auto log_event_idx = m_deserialized_log_events->size();
        SPDLOG_ERROR("Timestamp type is not int for log event index {}", log_event_idx);
    }
```